src/SupportInfillPart.h

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2017 Ultimaker B.V.
+// CuraEngine is released under the terms of the AGPLv3 or higher.
+
+#ifndef SUPPORT_INFILL_PART_H
+#define SUPPORT_INFILL_PART_H
+
+#include <vector>
+
+#include "utils/AABB.h"
+#include "utils/polygon.h"
+
+
+namespace cura
+{
+
+/*!
+ * A SupportInfillPart represents an connected area for support infill on a layer.
+ * The support infill areas on a layer can be isolated islands, and in this case, a SupportInfillPart represents a single island.
+ *
+ * This data structure is required for gradual support, which needs to partition a support area into a number of sub-areas with different density.
+ * Because support is handled as a whole in the engine, that is, we have a global support areas instead of support areas for each mesh.
+ * With this data structure, we can keep track of which gradual support infill areas belongs to which support area, so we can print them together.
+ */
+class SupportInfillPart
+{
+public:
+    Polygons outline;  //!< The outline of the support infill area
```

Comment:
Actually this should be of the type PolygonsPart.
That way we ensure that each SupportInfillPart is exactly one part.

That does mean that you will have to move the splitIntoSmallerParts out of this class, though.
You'll have to perform that functionality inside the the function where you are now calling splitIntoSmallerParts from.